Build failures on master

On macOS 10.14:
```
/Users/zach/builds/4f04be8b/0/turi/turicreate/src/capi/impl/capi_error_handling.cpp:7:
In file included from /Users/zach/builds/4f04be8b/0/turi/turicreate/src/capi/impl/capi_error_handling.hpp:9:
In file included from /Users/zach/builds/4f04be8b/0/turi/turicreate/src/capi/impl/capi_wrapper_structs.hpp:10:
/Users/zach/builds/4f04be8b/0/turi/turicreate/src/unity/lib/variant.hpp:11:10: fatal error: 'boost/variant.hpp' file not found
#include <boost/variant.hpp>
         ^~~~~~~~~~~~~~~~~~~
```

Also repros on linux

Introduced by #972

--- a/src/capi/CMakeLists.txt
+++ b/src/capi/CMakeLists.txt
@@ -32,7 +32,7 @@ add_library(_capi_objects OBJECT
 )
 
 set(_capi_requires unity unity_core numerics)
- 
+add_dependencies(_capi_objects ${_capi_requires})
 
 make_library(capi 
   SOURCES

